src/token/api/src/canister.rs

```diff
@@ -348,126 +315,489 @@ pub trait TokenCanisterAPI: Canister + Sized {
+    }
+
+    fn test_canister() -> TokenCanisterMock {
+        MockContext::new().with_caller(alice()).inject();
+
+        let canister = TokenCanisterMock::init_instance();
+        canister.init(
+            Metadata {
+                logo: "".to_string(),
+                name: "".to_string(),
+                symbol: "".to_string(),
+                decimals: 8,
+                owner: alice(),
+                fee: Tokens128::from(0),
+                fee_to: alice(),
+                is_test_token: None,
+            },
+            Tokens128::from(1000),
+        );
+
+        canister.state.borrow_mut().stats.min_cycles = 0;
+
+        canister
+    }
+
+    #[test]
+    fn transfer_to_same_account() {
+        let canister = test_canister();
+        let transfer = TransferArgs {
+            from_subaccount: None,
+            to: alice().into(),
+            amount: 100.into(),
+            fee: None,
+            memo: None,
+            created_at_time: None,
+        };
+
+        let res = canister.icrc1_transfer(transfer);
+        assert_eq!(
+            res,
+            Err(TransferError::GenericError {
+                error_code: 500,
+                message: "Self transfer".into()
+            })
+        )
+    }
+
+    #[test]
+    fn transfer_to_same_default_subaccount() {
+        let canister = test_canister();
+        let transfer = TransferArgs {
+            from_subaccount: Some(crate::account::DEFAULT_SUBACCOUNT),
+            to: alice().into(),
+            amount: 100.into(),
+            fee: None,
+            memo: None,
+            created_at_time: None,
+        };
+
+        let res = canister.icrc1_transfer(transfer);
+        assert_eq!(
+            res,
+            Err(TransferError::GenericError {
+                error_code: 500,
+                message: "Self transfer".into()
+            })
+        );
+
+        let transfer = TransferArgs {
+            from_subaccount: None,
+            to: Account::new(alice(), Some(DEFAULT_SUBACCOUNT)),
+            amount: 100.into(),
+            fee: None,
+            memo: None,
+            created_at_time: None,
+        };
+
+        let res = canister.icrc1_transfer(transfer);
+        assert_eq!(
+            res,
+            Err(TransferError::GenericError {
+                error_code: 500,
+                message: "Self transfer".into()
+            })
+        );
+    }
+
+    #[test]
+    fn test_claim() {
+        let bob_sub = gen_subaccount();
+        let alice_sub = gen_subaccount();
+
+        let alice_aid =
+            AccountIdentifier::new(alice().into(), Some(SubaccountIdentifier(alice_sub)));
+        let bob_aid = AccountIdentifier::new(bob().into(), Some(SubaccountIdentifier(bob_sub)));
+
+        let (ctx, canister) = test_context();
+        ctx.update_caller(john());
+
+        assert!(canister
+            .mint(
+                canister.owner(),
+                Some(alice_aid.to_address()),
```

Comment:
so, this would just be for default accounts?

Reply:
It works just like before - subaccounts can be used to specify any AccountIdentifier, but `claim` method transfers tokens to the default account of the claiming user.